pkg/distributor/distributor.go

```diff
@@ -77,6 +78,11 @@ var (
 	})
 )
 
+type clusterTracker interface {
+	CheckReplica(ctx context.Context, cluster string, replica string) error
+	Stop()
+}
```

Comment:
As the implementation is internal to this package, and we didn't go with the layered approach, probably don't need an interface for this.